src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_IndexerAccess.cs

```diff
@@ -117,6 +118,8 @@ private BoundExpression VisitIndexerAccess(BoundIndexerAccess node, bool isLeftO
                 // This is an indexer set access. We return a BoundIndexerAccess node here.
                 // This node will be rewritten with MakePropertyAssignment when rewriting the enclosing BoundAssignmentOperator.
 
+                arguments = unwrapPlaceholdersIfNeeded(arguments);
```

Comment:
:memo: Wasn't able to avoid unwrapping placeholders here because we're not visiting the arguments yet (in L-value scenario)

Reply:
> Wasn't able to avoid unwrapping placeholders here because we're not visiting the arguments yet (in L-value scenario)

Could you please provide more details about this scenario and the code path how we arrive here?

Reply:
I think I understand what is going on here and I think that the right way to deal with this is to avoid prematurely lowering ```BoundIndexOrRangePatternIndexerAccess``` to ```BoundIndexerAccess``` when it is a ```isLeftOfAssignment```. The same way as we do not lower ```BoundIndexerAccess``` to a method call. I'd like to try this out once this PR is merged.

Reply:
Filled [https://github.com/dotnet/roslyn/issues/57810](https://github.com/dotnet/roslyn/issues/57810)